src/store/eventing/mode.ts

```diff
@@ -0,0 +1,7 @@
+import eventManager from '../../common/EventManager';
+import { AppState } from '../types';
+import { Event } from '../../@types';
+
+export const handleToggleAnnotationModeAction = (prevState: AppState, nextState: AppState): void => {
+    eventManager.emit(Event.ANNOTATIONS_MODE_CHANGE, nextState.common.mode);
```

Comment:
Can we make this payload an object in case we need to add more properties to it later?